app/lib/service/openid/openssl_commands.dart

```diff
@@ -0,0 +1,145 @@
+        'genrsa',
+        '-out',
+        privateKeyPath,
+        '$keySize',
+      ],
+      timeout: Duration(seconds: 10),
+    );
+    if (pr1.exitCode != 0) {
+      throw Exception('Unable to run openssl:\n${pr1.asJoinedOutput}');
+    }
+    final privateContent = await File(privateKeyPath).readAsString();
+    final privateKeyBytes = decodePemBlocks(PemLabel.privateKey,
+            privateContent.replaceAll(' RSA PRIVATE KEY', ' PRIVATE KEY'))
+        .single;
+
+    final pr2 = await runProc(
+      [
+        'openssl',
+        'rsa',
+        '-in',
+        privateKeyPath,
+        '-pubout',
+        '-out',
+        publicKeyPath,
+      ],
+      timeout: Duration(seconds: 10),
+    );
+    if (pr2.exitCode != 0) {
+      throw Exception('Unable to run openssl:\n${pr1.asJoinedOutput}');
+    }
+    final publicContent = await File(publicKeyPath).readAsString();
+    final publicKeyBytes =
+        decodePemBlocks(PemLabel.publicKey, publicContent).single;
+    return RsaKeyPair(
+      Uint8List.fromList(privateKeyBytes),
+      Uint8List.fromList(publicKeyBytes),
+    );
+  });
+}
+
+/// Signs [input] text with the private RSA key and returns the signature bytes.
+@visibleForTesting
+Future<Uint8List> signTextWithRsa({
+  required String input,
+  required List<int> privateKey,
+}) async {
+  return await withTempDirectory((dir) async {
+    final outputFile = File(p.join(dir.path, 'output.sign'));
+    final inputFile = File(p.join(dir.path, 'input.txt'));
+    await inputFile.writeAsString(input);
+    final privateKeyFile = File(p.join(dir.path, 'private.pem'));
+    await privateKeyFile.writeAsString(
+        encodePemBlock(PemLabel.privateKey, privateKey)
+            .replaceAll(' PRIVATE KEY', ' RSA PRIVATE KEY'));
+    final pr = await runProc(
+      [
+        'openssl',
+        'dgst',
+        '-sha256',
+        '-binary',
+        '-sign',
+        privateKeyFile.path,
+        '-out',
+        outputFile.path,
+        inputFile.path,
+      ],
+      timeout: Duration(seconds: 10),
+    );
+    if (pr.exitCode != 0) {
+      throw Exception('Unable to run openssl:\n${pr.asJoinedOutput}');
+    }
+    return await outputFile.readAsBytes();
+  });
+}
+
+/// Verifies if [signature] is valid for [input], using the RSA public key.
+Future<bool> verifyTextWithRsaSignature({
+  required String input,
+  required List<int> signature,
+  required List<int> publicKey,
+}) async {
+  return await withTempDirectory((dir) async {
+    final inputFile = File(p.join(dir.path, 'input.txt'));
+    await inputFile.writeAsString(input);
+    final signatureFile = File(p.join(dir.path, 'input.sign'));
+    await signatureFile.writeAsBytes(signature);
+    final publicKeyFile = File(p.join(dir.path, 'public.pem'));
+    final publicPemContent = encodePemBlock(PemLabel.publicKey, publicKey);
+    await publicKeyFile.writeAsString(publicPemContent);
+    final pr = await runProc(
+      [
+        'openssl',
+        'dgst',
```

Comment:
We could also consider:
https://www.openssl.org/docs/manmaster/man1/openssl-pkeyutl.html

But I don't see how it would be much different.